Docs/readme (#516)
LnL7/nix-darwin: README.md

Nix modules for darwin, `/etc/nixos/configuration.nix` for macOS.

This project aims to bring the convenience of a declarative system approach to macOS.

> Nix-darwin is build up around [nixpkgs](https://github.com/NixOS/nixpkgs), quite similar to [NixOs](https://nixos.org/).

Reviewer: ```suggestion
Nix-darwin is built up around [Nixpkgs](https://github.com/NixOS/nixpkgs), quite similar to [NixOs](https://nixos.org/).
```

Nixpkgs tends to be capitalized in the manual.

## Install

> To install nix-darwin, a working installation of [nix](https://github.com/NixOS/nix#installation) is required.

Reviewer: ```suggestion
To install nix-darwin, a working installation of [Nix](https://github.com/NixOS/nix#installation) is required.
```

Nix is capitalized in the manual.

darwin-rebuild changelog
```

> > NOTE: If you are using nix as a daemon service the channel for that will be owned by root.
> > Use `sudo -i nix-channel --update darwin` instead.

Reviewer: ```suggestion
> NOTE: If you are using a multi-user install of Nix, which is now the default, the Nix daemon will use root's channels.
>These need to be updated as root and because of the configuration of `sudo` on macOS we need to pass the `-i` flag.
> Use `sudo -i nix-channel --update darwin` instead.
```

./result/bin/darwin-uninstaller
```

> > NOTE: This will also uninstall nix alongside nix-darwin.

Reviewer: ```suggestion
> NOTE: This will also uninstall Nix alongside nix-darwin.
```

$
```

## Documentation

Reference documentation of all the options is available [here](https://lnl7.github.io/nix-darwin/manual/index.html#sec-options).--
This can also be accessed locally using `man 5 configuration.nix`.

`darwin-help` will open a HTML version of the manpage in the default browser.

> Furthermore there's `darwin-option` to introspect the settings of a system and its available options.

Reviewer 2: Pretty sure that `darwin-option` does not work when using Flakes. I don’t have access to my Mac at the moment, so can’t confirm for sure though. If I’m right, maybe that should be mentioned here.

Author: That has also been reported elsewhere. Added a note right after.

$
```

## Documentation

> Reference documentation of all the options is available here
> https://lnl7.github.io/nix-darwin/manual/index.html#sec-options.

Reviewer: ```suggestion
Reference documentation of all the options is available [here](https://lnl7.github.io/nix-darwin/manual/index.html#sec-options).
```

This can also be accessed locally using `man 5 configuration.nix`.

> `darwin-help` will open a html version of the manpage in the default browser.

Reviewer: ```suggestion
`darwin-help` will open an HTML version of the manpage in the default browser.
```

a/an basically comes down to style but HTML should be capitalized.

> Furthermore there's `darwin-option` to introspect the settings of a system and it's available options.

Reviewer: ```suggestion
Furthermore there's `darwin-option` to introspect the settings of a system and its available options.
```

you can run them like this:

```bash

> > run all tests
> nix-build release.nix -A tests
> > or just a subset
> nix-build release.nix -A tests.environment-path

Reviewer: ```suggestion
# run all tests
nix-build release.nix -A tests
# or just a subset
nix-build release.nix -A tests.environment-path
```

Everything inside a bash code block should be valid bash.
